internal/provisioner/controller/gatewayclass.go

```diff
@@ -37,6 +38,11 @@ import (
 	gatewayapi_v1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 )
 
+const (
+	gatewayAPIBundleVersionAnnotation   = "gateway.networking.k8s.io/bundle-version"
+	gatewayAPICRDBundleSupportedVersion = "v1.0.0"
```

Comment:
we might run into issues when we bump the go module, trying to think of a good way to fix this beyond adding a constant to the upstream code we can import

Reply:
thinking that an e2e test might have to be the way we fix this, just make sure that by default the SupportedVersion condition is always set to true, if we miss updating this value but update the go module/regenerate example YAMLs the test would fail

Reply:
Yeah I think an E2E is a good idea to ensure we're in sync (even if upstream adds a constant, we all forget to bump things sometimes 😄)